server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsService.java

```diff
@@ -410,6 +532,118 @@ public void clusterChanged(ClusterChangedEvent event) {
                 }
             }
         }
+        if (currentMaster == null && clusterService.localNode().isMasterNode()) {
```

Comment:
I don't think we want to poll schedule polling all master eligible nodes on every cluster changed event and on every master eligible node.

I'd have thought, given how deep this step is in the diagnosis process we take it "on demand" (I suppose we can think about optimisations like caching/refresh and such but that's a different discussion)

@DaveCTurner what do you think?

Note that this _seems_ like we're scheduling several polling tracks (on every cluster changed event we begin a 10 seconds interval schedule https://github.com/elastic/elasticsearch/pull/88020/files#diff-6898ba5666bbb351e267e9295e8feb7fbabf0d95d77aa4375c06cb2c16ad0a7dR626 )

Reply:
Since all calls to `beginPollingClusterFormationInfo()` and `cancelPollingClusterFormationInfo()` are on the same thread (the cluster state changed thread), and we always call `cancelPollingClusterFormationInfo()` before starting the next batch of tasks in `beginPollingClusterFormationInfo()`, we never have multiple polling tracks active at the same time, do we? In the worst case we have some that are in the process of being canceled, but we ignore those after we request their cancellation. Is your fear that we'll build up a huge number of tasks that we've requested be canceled but haven't finished canceling yet?

Reply:
I think it's strange that `beginPollingClusterFormationInfo` starts by cancelling the polling. 
I guess the scenario it's trying to guard against is "node has master -> no master (0) -> node has master (1) -> no master (2)" where the 0 - 1- 2 events happen very quickly.
Is this the only scenario we're guarding against? 

What if we use some form of `Atomic...` entity to state if "polling is progress" and don't schedule another track if one is running?

Reply:
The polling logic was moved to its own PR (#88397) where we hashed out all of the polling-related conversations. So I'm resolving all of them here.